UnityProject/Assets/Scripts/Core/Editor/PropertyDrawers/HealthSystemBasePropertyDrawer.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Core.Editor.Attributes;
+using HealthV2.Living.PolymorphicSystems;
+using UnityEditor;
+using UnityEngine;
+
+[CustomPropertyDrawer(typeof(SelectImplementationAttribute))]
+public class HealthSystemBasePropertyDrawer : PropertyDrawer
+{
+    private Type[] _implementations;
+    private int _implementationTypeIndex;
+
+    public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
```

Comment:
This doesn't give human-readable names for implementations, and can get out of hand quickly.